packages/babel-types/src/validators/isReferenced.js

```diff
@@ -66,6 +67,9 @@ export default function isReferenced(
       if (parent.value === node) {
         return !grandparent || grandparent.type !== "ObjectPattern";
       }
+      if (parent.params && parent.params.indexOf(node) !== -1) {
```

Comment:
```suggestion
      if (parent.params.includes(node)) {
```

Reply:
@nicolo-ribaudo you asked me to remove the use of `.includes` for the sake of Node 6 support, but now that I check, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes indicates it's supported there. Is using `.includes` ok?

Reply:
Oh I thought that it wasn't 😅

If it works, it's ok to use it.

Reply:
Cool, I checked a node 6 repl and `.includes` seems to work.

@jridgewell removing the check for whether `parent.params` is present before calling `.includes`, does mean an error would be thrown if we reach this line when parent is an `ObjectProperty` or `ClassProperty`. The only way that could happen is if `node` is neither the key nor the value of the parent Object/ClassProperty, which I think currently could only happen if it's a decorator or type annotation. I don't know whether it's important for `isReferenced` to be able to handle being called with non-identifier-like nodes such as Decorator.

Reply:
What about refactoring the switch like this then?

```js
case "ClassMethod":
case "ClassPrivateMethod":
case "ObjectMethod":
  check parent.params.includes(node);
  /* falls through */
case "ObjectProperty":
case "ClassProperty":
case "ClassPrivateProperty":
  // all the existing checks
```

Reply:
That works, updated.

I'm now getting some failing tests in babel-preset-env related to "edge" being 18 instead of the fixture's expectation of 17, but I get them on master too - I'm guessing it was caused by a new version of caniuse-lite being released today.

Reply:
Yeah ignore those tests